pdm/src/main/java/me/bristermitten/pdm/PDMBuilder.java

```diff
@@ -34,6 +38,38 @@ public PDMBuilder(@NotNull final Plugin plugin)
         applicationVersion(plugin.getDescription().getVersion());
     }
 
+    public PDMBuilder(@NotNull final Class<? extends Plugin> plugin)
+    {
+        Validate.isTrue("org.bukkit.plugin.java.PluginClassLoader".equals(plugin.getClassLoader().getClass().getName()),
+            "Plugin must be loaded with a PluginClassLoader");
+        classLoader((URLClassLoader) plugin.getClassLoader());
+        PluginDescriptionFile description = get(classLoader, "description");
+        dependenciesResource(classLoader.getResourceAsStream(DEPENDENCIES_RESOURCE_NAME));
+        rootDirectory(new File("./plugins"));
+        applicationName(description.getName());
+        applicationVersion(description.getVersion());
+        loggerFactory(clazz -> Logger.getLogger(description.getName()));
```

Comment:
Yeah you're right, `JavaPlugin`s use `PluginLogger`s. Not sure how we could use that here though, because when the logger factory is called the plugin instance may and most likely will not exist yet

Reply:
Indeed. We should probably keep this as-is for now, if it causes any significant problems we can rethink.